### PR TITLE
Refactor Attach.exe with System.CommandLine

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
Add System.CommandLine dependency, support parsing multiple ConfigFiles, support Engine settings

### CODE REVIEW

File: src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+    [Flags]
+    internal enum LoggingFlags
+    {
+        None = 0,
+        Errors = 1,
+        Messages = 2,
+        Dumps = 4
+    }
+
+    internal class AttachCommandHandler
```
Comment (wiktork) [inline src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs]: >AttachCommandHandler [](start = 19, length = 20)

Any significant changes in the body here or is it mostly a refactor?

File: src/InstrumentationEngine.Attach/InstrumentationEngine.Attach.csproj [hunk 2]
```diff
@@ -32,17 +32,18 @@
...
   <ItemGroup>
-    <EmbeddedResource Include="InstrumentationConfigurationSources.xsd" />
-    <EmbeddedResource Include="EngineConfiguration.xsd" />
+    <EmbeddedResource Include="Schema\InstrumentationConfigurationSources.xsd" />
+    <EmbeddedResource Include="Schema\EngineConfiguration.xsd" />
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Diagnostics.NETCore.Client" Version="0.2.61701" />
     <PackageReference Include="Microsoft.VisualStudioEng.MicroBuild.Core" Version="$(MicroBuildCoreVersion)" />
+    <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20104.2" />
```
Comment (wiktork) [inline src/InstrumentationEngine.Attach/InstrumentationEngine.Attach.csproj]: >CommandLine [](start = 38, length = 11)

How much size overhead does this add to our distribution? This makes a lot of sense for command line tools (i.e. dotnet tools all use this library) but this seems unnecessary for something that isn't spawned directly by the user.

File: src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs (hunk 1 above)
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs]: Mostly a refactor, but also added multiple configs and engine logging parameters

---
In reply to: [394138338](https://github.com/microsoft/CLRInstrumentationEngine/pull/234#discussion_r394138338) [](ancestors = 394138338)

File: src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+                    WriteError(Invariant($"Could not find directory '{configFileDirectory}'."));
+                    return ExitCodeFailure;
+                }
+
+                sourceInfo.ConfigSourceDirectory = configFileDirectory;
+
+                #region Parse Configuration Sources
+
+                // Schema file embedded name is the same that the full type name that was generated from the schema file
+                string schemaResourceName = typeof(InstrumentationConfigurationSources).FullName + ".xsd";
```
Comment (WilliamXieMSFT) [inline src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs]: >string schemaResourceName = typeof(InstrumentationConfigurationSources).FullName + ".xsd"; [](start = 16, length = 90)

moving this before the foreach loop

File: src/InstrumentationEngine.Attach/InstrumentationEngine.Attach.csproj (hunk 2 above)
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine.Attach/InstrumentationEngine.Attach.csproj]: The difference in size for the extracted Release builds is about 150 KB (from 4.1 MB to 4.25 MB). I think the benefits of parameter validation and help text are worth the small size growth

---
In reply to: [394139546](https://github.com/microsoft/CLRInstrumentationEngine/pull/234#discussion_r394139546) [](ancestors = 394139546)

**Comment by WilliamXieMSFT** [comment]
> <?xml version="1.0" encoding="utf-8"?>

accidentally added this file, removing

---
Refers to: tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.vcxproj:1 in c3da2c2. [](commit_id = c3da2c2007d7741d980769e0ed08458db6929c91, deletion_comment = False)

**Pull request merged**